Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/ArrayDictionary.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArrayDictionary implements Dictionary {
+	private ArrayList <String> words=new ArrayList<String>();
+
+	//------------------------------------------
+	public ArrayDictionary(String filename){
+		try (BufferedReader r = new BufferedReader (new FileReader (filename));){
+			String temp = "";
+			do{
+				temp= r.readLine();
+				words.add(temp);
+			}while(temp!=null);
+		} catch (FileNotFoundException e) {
```

Comment:
I think maybe it would be better to not catch these exceptions and instead make them throwable by the class, then you can handle them in a more appropriate place, like the user interface.